### PR TITLE
Allow global struct definitions.

### PR DESCRIPTION
**Pull request by ekpyron**
Hm... I'm actually trying to break this, but so far I haven't found anything that *doesn't* work with these minimal changes...

### CODE REVIEW

**Comment by chriseth** [comment]
We could extend this PR by enums, too, but perhaps it would get too large then? :)

File: libsolidity/interface/CompilerStack.cpp [hunk 1]
```diff
@@ -285,17 +285,16 @@ bool CompilerStack::analyze()
 		// the special variables "this" and "super" must be set appropriately.
 		for (Source const* source: m_sourceOrder)
 			for (ASTPointer<ASTNode> const& node: source->ast->nodes())
+			{
+				if (!resolver.resolveNamesAndTypes(*node)) return false;
```
Comment (erak) [inline libsolidity/interface/CompilerStack.cpp]: Newline? ;-)

Comment (ekpyron) [reply libsolidity/interface/CompilerStack.cpp]: In my defense: it didn't have one before ;-).

**Comment by ekpyron** [comment]
I'll probably let this depend on https://github.com/ethereum/solidity/pull/7271, which will make it much easier to check, if this works in the presence of imports.
Ans once structs work, it should be trivial to extend this PR to enum's as well :-).

**Comment by ekpyron** [comment]
I was distracted by things like https://github.com/ethereum/solidity/pull/7271 too much to finish this in the end - if anyone wants to take over while I'm away, feel free - it might really be that this just works without much more changes and https://github.com/ethereum/solidity/pull/7271 (one it arrived at develop_060) should allow for the easy writing of tests for it in the potentially problematic import scenarios.

**Comment by leonardoalt** [comment]
https://github.com/ethereum/solidity/issues/6989

**Comment by ekpyron** [comment]
I wonder whether this should have some documentation changes, but I haven't found any mention of valid top-level constructs or something like that, but I might be missing it.

**Comment by ekpyron** [comment]
Ah, I still need to check the effects on the json AST.
EDIT: added some tests for it.

**Comment by chriseth** [comment]
Test failure - maybe indentation in the new test files?

**Comment by ekpyron** [comment]
The test failure is actually due to the documentation change for replacing the address cast in external functions... I'll create a PR for fixing that.

**Comment by ekpyron** [comment]
Actually, I think this can't be fixed and is due to the use of "address" as keyword - we should disable the solhint run for the 060 branch.

**Review by chriseth — APPROVED** [review]

**Pull request merged**